File: 프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.java
class Solution {
    public String solution(String my_string, String overwrite_string, int s) {
        String answer = "";
        String subs = my_string.substring(s, s+overwrite_string.length());
        
        int k=0;
        for(int i=0; i < my_string.length(); i++){
            if(i >= s && i < s+overwrite_string.length())
                answer += overwrite_string.charAt(k++);    
            else
                answer += my_string.charAt(i);    
        }
        return answer;
    }
}

File: 프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.kt
class Solution {
    fun solution(my_string: String, overwrite_string: String, s: Int): String {
        var answer: String = ""
        val len = overwrite_string.length

        var x = 0;
        my_string.toCharArray().forEachIndexed { index, c ->
            answer += if (index >= s && x < len) overwrite_string[x++]
            else c

        }

        return answer
    }
}